### PR TITLE
Remove transformations for older browser versions

### PR DESCRIPTION
**Pull request by Gudahtt**
The Babel config had previously supported all browsers with greater than 0.25% global usage (according to `browserlist`). This resulted in `babel-preset-env` including plugins sufficient to support the following minimum browser versions:

```
{
  "chrome": "49",
  "android": "4.4",
  "edge": "16",
  "firefox": "52",
  "ios": "9.3",
  "safari": "11"
}
```

Instead, the babel config now explicitly supports chrome >= 58 and firefox >= 53. Chrome and Firefox are the only browsers we currently publish to, and these were the minimum versions with no additional Babel transformations.

The minimum browser versions we support should be re-evaluated later, when we have added tests and documentation.

The plugin 'transform-async-to-generator' has also been removed. It was used to translate async/await, but our browser targets all support async/await.

Removing some of these transformations exposed bugs in `uglify-es` that only presented themselves in the production build. `gulp-uglify-es` has been updated to a version that uses `terser` instead of `uglify-es`, which has resolved these issues.

Relates to MetaMask#6805

### CODE REVIEW

**Comment by Gudahtt** [comment]
Oops... deleted the wrong branch 🤦‍♂

**Comment by Gudahtt** [comment]
I've [explained here](https://github.com/MetaMask/metamask-extension/issues/6805#issuecomment-508891738) why I've chosen Chrome 58 and Firefox 53.

We should probably gather more information and opinions before merging this PR. I feel pretty good about it, but if we discover a lot of users rely upon versions older than this, we might want to investigate more first before breaking the extension for them.

**Comment by danjm** [comment]
Need to see if we can get more usage info regarding browser versions.

**Comment by Gudahtt** [comment]
I just left [a comment on the issue](https://github.com/MetaMask/metamask-extension/issues/6805#issuecomment-509647940) describing the usage metrics I found.

tl;dr: between 0.00026%-0.047% of users would be negatively affected by the change in this PR (assuming that the metrics collected between July 1st-July 8th are representative). Given the extremely small set of affected users, merging this PR should be pretty safe even before deciding upon which browser versions to officially support.

**Review by whymarrh — COMMENTED** [review]
I'm not sure what's the best process here but I'm definitely in favour of this

**Review by whymarrh — APPROVED** [review]
LGTM

**Comment by kumavis** [comment]
would be nice if we could display an error screen if the user's browser doesnt match the requirements
or utilize a parameter in the manifest if there is one

**Comment by Gudahtt** [comment]
I haven't experimented with this yet, but we could set a minimum browser version in the manifest as well. That should prevent users on older browsers from updating/installing the extension to an incompatible version.

**Comment by whymarrh** [comment]
> [...] we could set a minimum browser version in the manifest as well. That should prevent users on older browsers from installing the extension on incompatible browsers, or updating it if they already have it installed.

If this is possible it's certainly worth doing (maybe its own PR?)

**Comment by Gudahtt** [comment]
I've added the minimum browser version to the manifest in #6877
